### PR TITLE
Front-end bug fixes and ESLint setup

### PR DESCRIPTION
**Pull request by alvyn279**
- Fixed missing price in orderbook, crash on scroll, and two decimal price
- Refactored the handling and spreading of the raw orderbook data to make it less costly on re-renders
- Reformatted all the front-end source code to satisfy new custom coding style

For clarity regarding the bug fixes, only view changes to the first commit.

### CODE REVIEW

File: app/src/styles/PriceLevel.js [hunk 1]
```diff
@@ -1,33 +1,35 @@
...
+        alignContent: 'flex-start',
+        padding: '5px 0px',
+        margin: '0px 10px',
+    },
+    price: {
+        float: 'left',
+        width: '50px',
+        textAlign: 'left',
+        paddingRight: 15,
+        color: Colors.darkGrey,
```
Comment (panoreak) [inline app/src/styles/PriceLevel.js]: nice 👍

File: app/.eslintrc.json [hunk 2]
```diff
@@ -8,14 +8,36 @@
...
+        "array-callback-return": "off",
+        "import/prefer-default-export": "off",
+        "no-unused-vars": ["error", { "vars": "all", "args": "none"}],
+        "no-unused-expressions": "off",
+        "no-loop-func": "off",
+        "no-plusplus": "off",
+        "radix": ["error", "as-needed"],
+        "react/jsx-max-props-per-line": [1, { "maximum": 1 }],
+        "react/prop-types": "off",
+        "react/jsx-filename-extension": "off",
```
Comment (chimano) [inline app/.eslintrc.json]: I like having the distinction between component files and other js files. Any reason why this was turned off?

File: app/.eslintrc.json [hunk 3]
```diff
@@ -8,14 +8,36 @@
...
+        "react/jsx-indent" : ["error", 4],
+        "react/jsx-indent-props" : ["error", 4],
+        "react/jsx-curly-brace-presence": ["error", { "props": "always", "children": "ignore"}],
+        "react/jsx-props-no-spreading": "off",
```
Comment (chimano) [inline app/.eslintrc.json]: Not a fan of this being off, can it be a warning instead. using the spread operator for props makes code harder to follow.

File: app/src/actions/actions.js [hunk 4]
```diff
@@ -1,8 +1,6 @@
...
-    return {
-        type: ACTION_TYPES.SAVE_REACT_APP_NAME,
-        payload: name
-    };
-};
\ No newline at end of file
+export const saveReactAppName = name => ({
+    type: ACTION_TYPES.SAVE_REACT_APP_NAME,
+    payload: name,
+});
```
Comment (chimano) [inline app/src/actions/actions.js]: We don't need this anymore

Comment (alvyn279) [reply app/src/actions/actions.js]: @chimano actually I want to leave this to demonstrate how I want the connection with redux to be done for future store interactions.

Comment (chimano) [reply app/src/actions/actions.js]: Fair enough

File: app/.eslintrc.json (hunk 3 above)
Comment (chimano) [reply app/.eslintrc.json]: Nevermind then just turn it off

Comment (alvyn279) [reply app/.eslintrc.json]: I agree with being strict with this rule, that's why i'm gonna just leave it as warning/error.

**Pull request merged**